docs/orleans/grains/grain-references.md

```diff
@@ -0,0 +1,214 @@
+
+Since a grain reference represents the logical identity of the target grain, it is independent of the physical location of the grain, and stays valid even after a complete restart of the system. Developers can use grain references like any other .NET object. It can be passed to a method, used as a method return value, etc., and even saved to persistent storage.
+
+A grain reference can be obtained by passing the identity of a grain to the <xref:Orleans.IGrainFactory.GetGrain%60%601(System.Type,System.Guid)?displayProperty=nameWithType> method, where `T` is the grain interface and `key` is the unique key of the grain within the type.
+
+The following are examples of how to obtain a grain reference of the `IPlayerGrain` interface defined above.
+
+From within a grain class:
+
+```csharp
+Guid playerId = Guid.NewGuid(); // This would typically be read from an HTTP request parameter or elsewhere.
+IPlayerGrain player = GrainFactory.GetGrain<IPlayerGrain>(playerId);
+```
+
+From Orleans client code:
+
+```csharp
+Guid playerId = Guid.NewGuid(); // This would typically be read from an HTTP request parameter or elsewhere.
+IPlayerGrain player = client.GetGrain<IPlayerGrain>(playerId);
+```
+
+Grain references contain three pieces of information:
+
+1. The grain _type_, which uniquely identifies the grain class.
+2. The grain _key_, which uniquely identifies a logical instance of that grain class.
+3. The _interface_ which the grain reference must implement.
+
+> [!NOTE]
+> The grain _type_ and _key_ form the [grain identity](grain-identity.md).
+
+Notice that the above calls to <xref:Orleans.IGrainFactory.GetGrain*?displayProperty=nameWithType> accepted only two of those three things:
+
+* The _interface_ implemented by the grain reference, `IPlayerGrain`.
+* The grain _key_, which is the value of `playerId`.
+
+Above, we said that a grain reference contains a grain _type_, _key_, and _interface_ but we have only provided Orleans with the _key_ and _interface_ in our example code. That is because Orleans maintains a mapping between grain interfaces and grain types. When you ask the grain factory for `IShoppingCartGrain`, Orleans consults its mapping to find the corresponding grain type so that it can create the reference. This works when there is only one implementation of a grain interface, but if there are multiple implementations, then you will need to disambiguate them in the `GetGrain` call. For more information, see the next section, [disambiguating grain type resolution](#disambiguating-grain-type-resolution).
```

Comment:
```suggestion
Despite stating that a grain reference contains a grain _type_, _key_, and _interface_, the examples only provided Orleans with the _key_ and _interface_. That is because Orleans maintains a mapping between grain interfaces and grain types. When you ask the grain factory for `IShoppingCartGrain`, Orleans consults its mapping to find the corresponding grain type so that it can create the reference. This works when there's only one implementation of a grain interface, but if there are multiple implementations, then you will need to disambiguate them in the `GetGrain` call. For more information, see the next section, [disambiguating grain type resolution](#disambiguating-grain-type-resolution).
```